homeassistant/components/bond/fan.py

```diff
@@ -0,0 +1,120 @@
+"""Support for Bond fans."""
+from typing import Any, Callable, Dict, List, Optional
+
+from bond import BOND_DEVICE_TYPE_CEILING_FAN, Bond
+
+from homeassistant.components.fan import (
+    SPEED_HIGH,
+    SPEED_LOW,
+    SPEED_MEDIUM,
+    SPEED_OFF,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+
+from ...const import ATTR_NAME
+from .const import DOMAIN
+from .utils import BondDevice, get_bond_devices
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up Bond cover devices."""
+    bond: Bond = hass.data[DOMAIN][entry.entry_id]
+
+    devices = await hass.async_add_executor_job(get_bond_devices, hass, bond)
+
+    fans = [
+        BondFan(bond, device)
+        for device in devices
+        if device.type == BOND_DEVICE_TYPE_CEILING_FAN
+    ]
+
+    async_add_entities(fans, True)
+
+
+class BondFan(FanEntity):
+    """Representation of a Bond cover."""
```

Comment:
some stale comments here refering to cover

Reply:
thanks! updated.